.github/workflows/dockerhub-latest-image.yml

```diff
@@ -6,6 +6,8 @@ on:
 jobs:
   publish-image-to-dockerhub:
     name: publish to DockerHub
+    permissions:
+      id-token: write # To be able to get OIDC ID token to sign images.
```

Comment:
Can you help to explain in more detail about the id-token? 

I can get some info from [cosign-installer usage](https://github.com/sigstore/cosign-installer#usage):
> This action does not need any GitHub permission to run, however, if your workflow needs to update, create or perform any action against your repository, then you should change the scope of the permission appropriately.

For example, if you are using the gcr.io as your registry to push the images you will need to give the write permission to the packages scope.

Reply:
Need the permissions of id-token to auto get the OIDC Identity Tokens from github action if we use keyless signatures.It would be failed if github action have not this permissions.

It's not necessary if we sign with cert or use other OIDC provider.

Here is the doc: https://docs.sigstore.dev/cosign/keyless/#identity-tokens

```shell
...
+ cosign sign --yes -a run_id=4913600087 ***/karmada-operator:cosign
Generating ephemeral keys...
Retrieving signed certificate...
Non-interactive mode detected, using device flow.
Enter the verification code MSBL-QTPQ in your browser at: https://oauth2.sigstore.dev/auth/device?user_code=MSBL-QTPQ
Code will be valid for 300 seconds
Error: signing [***/karmada-operator:cosign]: getting signer: getting key from Fulcio: retrieving cert: error obtaining token: expired_token
main.go:74: error during command execution: signing [***/karmada-operator:cosign]: getting signer: getting key from Fulcio: retrieving cert: error obtaining token: expired_token
```